src/TLabel.cpp

```diff
@@ -105,6 +107,12 @@ void TLabel::mouseDoubleClickEvent(QMouseEvent* event)
 
 void TLabel::mouseReleaseEvent(QMouseEvent* event)
 {
+    auto labelParent = dynamic_cast<TConsole*>(parent());
```

Comment:
May I suggest `qobject_cast` instead of `dynamic_cast` - it is better in Qt applications for several reasons - but is limited to objects (like the one here) that derive from the `QObject`base class...
* https://forum.qt.io/topic/59498/qobject_cast-vs-dynamic_cast
* https://stackoverflow.com/q/43994584/4805858

*Note, this also applies to the corresponding line in the `TTextEdit` changes.*